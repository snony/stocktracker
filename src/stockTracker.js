import React from 'react'
import { connect } from 'react-redux'
import InputSearchContainer from './components/search/form'
import './index.css'
import { ChartContainer, News, OverView, Peers, Stats } from './components/company/index'
import { mapDispatchToProps, mapStateToProps } from './redux/index'

const ComponentTitle = ({ title }) => <h3 className="component-title">{title}</h3>

export class StockTracker extends React.Component {
  componentDidMount() {
    this.props.getCompanies()
  }

  render() {
    const { news, overview, peers, keystats } = this.props.companyInfo
    return (
      <div className="stock-tracker-container">
        <InputSearchContainer />
        <ComponentTitle title="Historical Data" />
        <ChartContainer />
        <ComponentTitle title="News" />
        <News news={news} />
        <ComponentTitle title="Overview" />
        <OverView overview={overview} />
        <ComponentTitle title="Peers" />
        <Peers peers={peers} />
        <ComponentTitle title="Key Stats" />
        <Stats keystats={keystats} />
      </div>
    )
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(StockTracker)
