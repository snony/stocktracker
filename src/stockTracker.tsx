import * as React from 'react';

import History from './history'
import KeyStats from './keystats'
import News from './news'
import Overview from './overview'
import Peers from './peers'
import Search from './search'

import AdaptiveLogo from './logo'

const ComponentTitle = ({ title }: { title: string }) => <h3 className="component-title">{title}</h3>

export class StockTracker extends React.Component<{}, {}>{

  public render() {
    return (
      <div className="stock-tracker-container">
        <div className="stock-tracker-container__logo">
          <AdaptiveLogo />
        </div>
        <div className="stock-tracker-container__search">
          <Search />
        </div>
        <div className="stock-tracker-container__history">
          <History />
        </div>
        <div className="stock-tracker-container__news">
          <ComponentTitle title="News" />
          <News />
        </div>
        <div className="stock-tracker-container__key-stats">
          <ComponentTitle title="Key Stats" />
          <KeyStats />
        </div>
        <div className="stock-tracker-overview">
          <ComponentTitle title="Overview" />
          <Overview />
        </div>
        <div className="stock-tracker-container__peers">
          <ComponentTitle title="Peers" />
          <Peers />
        </div>
      </div>)
  }
}

export default StockTracker
