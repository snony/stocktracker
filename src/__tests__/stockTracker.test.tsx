import { shallow } from 'enzyme'
import React from 'react'

import { StockTracker } from '../stockTracker'

describe('StockTracker Component', () => {
  it('should render correctly', () => {
    const stockTrackerWrapper = shallow(<StockTracker />)
    expect(stockTrackerWrapper).toMatchSnapshot()
  })
})
