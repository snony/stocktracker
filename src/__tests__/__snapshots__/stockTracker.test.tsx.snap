// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockTracker Component should render correctly 1`] = `
<div
  className="stock-tracker-container"
>
  <div
    className="stock-tracker-container__logo"
  >
    <AdaptiveLogo />
  </div>
  <div
    className="stock-tracker-container__search"
  >
    <Connect(Search) />
  </div>
  <div>
    <Connect(Quote) />
  </div>
  <div
    className="stock-tracker-container__history"
  >
    <Connect(History) />
  </div>
  <div
    className="stock-tracker-container__news"
  >
    <ComponentTitle
      title="News"
    />
    <Connect(NewsItems) />
  </div>
  <div
    className="stock-tracker-container__key-stats"
  >
    <ComponentTitle
      title="Key Stats"
    />
    <Connect(KeyStats) />
  </div>
  <div
    className="stock-tracker-overview"
  >
    <ComponentTitle
      title="Overview"
    />
    <Connect(Overview) />
  </div>
  <div
    className="stock-tracker-container__peers"
  >
    <ComponentTitle
      title="Peers"
    />
    <Connect(Peers) />
  </div>
</div>
`;
